lib/galaxy/workflow/steps.py

```diff
@@ -33,6 +33,19 @@ def order_workflow_steps(steps):
         if not step.position or "left" not in step.position or "top" not in step.position:
             position_data_available = False
     if position_data_available:
+        # find minimum left and top values to normalize position
+        min_left = min(step.position["left"] for step in steps)
+        min_top = min(step.position["top"] for step in steps)
+        # normalize by min_left and min_top
+        for step in steps:
+            step.position["left"] -= min_left
+            step.position["right"] -= min_left
+            step.position["x"] -= min_left
+            step.position["top"] -= min_top
+            step.position["bottom"] -= min_top
+            step.position["y"] -= min_top
```

Comment:
Are all these attributes (apart from top and left) really necessary? I couldn't find where in the code they are being set.

Reply:
They're not, we should drop them IMO.